src/store.c

```diff
@@ -79,29 +88,80 @@ struct rill_store
 // -----------------------------------------------------------------------------
 
 static struct encoder store_encoder(
-        struct rill_store *store, struct indexer *indexer)
+        struct rill_store *store,
+        struct indexer *indexer,
+        struct vals* vals,
+        uint64_t offset)
 {
     return make_encoder(
-            store->vma + store->head->data_off,
+            store->vma + offset,
             store->vma + store->vma_len,
-            store->vals, indexer);
+            vals,
+            indexer);
 }
 
-static struct decoder store_decoder(struct rill_store *store)
+static struct decoder store_decoder(
+        struct rill_store *store, enum rill_column_e column)
 {
+    assert(column == rill_column_a || column == rill_column_b);
+
+    size_t offset = 0;
+    size_t offset_end = 0;
+    struct index* lookup = NULL;
+    struct index* index = NULL;
+
+    if (column == rill_column_a) {
```

Comment:
I think the function would be a bit simpler, if the call to make_decoder was in the if statement itself as it would avoid having to create a bunch of temporary variables and the common logic is minimal and unlikely to cause problems.